src/main/core/main.rs

```diff
@@ -142,8 +142,6 @@ pub fn run_shadow(build_info: &ShadowBuildInfo, args: Vec<&OsStr>) -> anyhow::Re
 
     // start up the logging subsystem to handle all future messages
     shadow_logger::init(log_level.to_level_filter()).unwrap();
-    // register the C logger
-    unsafe { logger::logger_setDefault(c::rustlogger_new()) };
```

Comment:
The new `log_c2rust` crate doesn't have a `bindgen` step to expose its APIs to Rust code. I don't think it makes sense to add this step, since it's really intended only for use by C code.

We also won't need explicit initialization after folding `logger` into this crate, at which point `logger` will always use `rustlogger`  back-end.